contracts/src/v0.8/EIP_712_1014_4337/interfaces/IAccount.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.12;
+
+import "./UserOperation.sol";
+
+interface IAccount {
+
+    /**
+     * Validate user's signature and nonce
+     * the entryPoint will make the call to the recipient only if this validation call returns successfully.
+     * signature failure should be reported by returning SIG_VALIDATION_FAILED (1).
+     * This allows making a "simulation call" without a valid signature
+     * Other failures (e.g. nonce mismatch, or invalid signature format) should still revert to signal failure.
```

Comment:
This makes the SCA somewhat tricky to implement (I don't think your SCA.sol adheres to these requirements). Is there a standard SCA as part of EIP-4337 that implements this interface?

Reply:
Right now the contract just returns 0 and does not use a paymaster or aggregator. Those are plugins that can augment the current experience. I'll see if other SCAs are currently used.

Reply:
Here's an example event on the Goerli contract:
https://goerli.etherscan.io/tx/0x5a894b1200bb65638d6802909fdfaf7dd314453e4f3c300abb4a613fc38c247c#eventlog

So far all the SCAs (senders) I see being used are unverified.

Reply:
Any idea what the difference between "signature failure" and nonce mismatch is? Doesn't a nonce mismatch cause a signature failure?

> signature failure should be reported by returning SIG_VALIDATION_FAILED (1)

> Other failures (e.g. nonce mismatch, ..) should still revert to signal failure